src/System.Memory/tests/Performance/Perf.ReadOnlySequence.First.cs

```diff
@@ -0,0 +1,367 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Memory.Tests;
+using System.MemoryTests;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Buffers.Tests
+{
+    public class Rerf_ReadOnlySequence_First
+    {
+        private const int InnerCount = 1000;
+        volatile static int _volatileInt = 0;
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void Byte_Array(int size, int start, int lenght)
+        {
+            var buffer = new ReadOnlySequence<byte>(new byte[size], start, lenght);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < size / 10; j++)
+                        {
+                            ReadOnlyMemory<byte> first = buffer.First;
+                            localInt ^= first.Length;
+                        }
+                    }
+                }
+                _volatileInt = localInt;
+            }
+        }
+
+        [Benchmark(InnerIterationCount = InnerCount)]
+        [InlineData(1000, 0, 1000)]
+        [InlineData(1000, 100, 800)]
+        [InlineData(1000 * 1000, 0, 1000 * 1000)]
+        [InlineData(1000 * 1000, 1000, 998 * 1000)]
+        private static void Byte_Memory(int size, int start, int lenght)
+        {
+            var manager = new CustomMemoryForTest<byte>(new byte[size], start, lenght);
+            var buffer = new ReadOnlySequence<byte>(manager.Memory);
+
+            foreach (BenchmarkIteration iteration in Benchmark.Iterations)
+            {
+                int localInt = 0;
+                using (iteration.StartMeasurement())
+                {
+                    for (int i = 0; i < Benchmark.InnerIterationCount; i++)
+                    {
+                        for (int j = 0; j < size / 10; j++)
```

Comment:
Why ``size/10`?

Reply:
MemoryManager is slow so I make less loops to reduce test time.

Reply:
In that case, change the value of Benchmark.InnerIterationCount for this test.

```C#
[Benchmark(InnerIterationCount = InnerCount / 10)]
```